deepmd/entrypoints/test.py

```diff
@@ -563,19 +564,31 @@ def test_dipole(
         munber of tests to do
     detail_file : Optional[str]
         file where test details will be output
+    has_atom_dipole : bool
+        whether atomic dipole is provided
 
     Returns
     -------
     Tuple[List[np.ndarray], List[int]]
         arrays with results and their shapes
     """
     data.add(
-        "dipole", 3, atomic=True, must=True, high_prec=False, type_sel=dp.get_sel_type()
+        "dipole", 3, atomic=has_atom_dipole, must=True, high_prec=False, type_sel=dp.get_sel_type()
     )
     test_data = data.get_test()
     dipole, numb_test, _ = run_test(dp, test_data, numb_test)
+
+    # do summation in atom dimension
+    if has_atom_dipole == False:
+        dipole = np.reshape(dipole,(dipole.shape[0], -1, 3))
+        atoms = dipole.shape[1]
+        dipole = np.sum(dipole,axis=1)
+
     l2f = l2err(dipole - test_data["dipole"][:numb_test])
 
+    if has_atom_dipole == False:
+        l2f = l2f / atoms
+
```

Comment:
The design here is not good.

We assume that `has_atom_dipole == False`, then by line 576, the label is ensured to be global. The atoms on line 584 is always 1.

When `has_atom_dipole == False`, the user wants to test the error in global dipole. The RMSE should be normalized by number of atoms. This information can be get from `test_data`.

Reply:
I don't think it's true. Suppose has_atom_dipole is false.
The frozen model will output local dipole. Therefore, it should be shaped as (nframs x sel_atom x 3).
However, the label is (nframes x 3) in this case.
Therefore, it's necessary to store natoms and do normalization

Reply:
That's clear to me